### PR TITLE
Reconcile

### PR DESCRIPTION
**Pull request by alecmerdler**
### Description

Deploy all Kubernetes objects when a `QuayRegistry` is created. Deployed objects are cleaned up by native k8s garbage collection using `ownerReferences`.

### Testing

**E2E Tests:**

```sh
$ go test -v ./controllers
```

**Manual:**

1. In one terminal, run the Operator locally:
```sh
$ go run main.go
```

2. In another terminal, create your base Quay config `Secret`:
```sh
$ kubectl create secret -n <your-namespace> generic --from-file=config.yaml=./kustomize/app/bundle/config.yaml quay-config-secret-123abc
```

3. Create your `QuayRegistry`:
```sh
$ kubectl create -n skynet-registry -f ./config/samples/quay.redhat.com_v1_quayregistry.yaml
```

4. Watch the logs from the Operator process to see the objects being created. Check your Kubernetes cluster to confirm everything was deployed.

5. For teardown, simply delete the `QuayRegistry`. Everything will be cleaned up automatically by k8s garbage collection using `ownerReferences`.

Fulfills https://issues.redhat.com/browse/PROJQUAY-827

### CODE REVIEW

File: api/v1/groupversion_info.go [hunk 1]
```diff
@@ -26,7 +26,7 @@ import (
 
 var (
 	// GroupVersion is group version used to register these objects
-	GroupVersion = schema.GroupVersion{Group: "quay.redhat.com.quay.redhat.com", Version: "v1"}
+	GroupVersion = schema.GroupVersion{Group: "quay.redhat.com", Version: "v1"}
```
Comment (josephschorr) [inline api/v1/groupversion_info.go:29]: If the intention is that this is a project quay project, we likely shouldn't use `redhat.com` in the group version :)

File: controllers/quayregistry_controller.go [hunk 2]
```diff
@@ -36,12 +42,53 @@ type QuayRegistryReconciler struct {
...
+	}
+
+	var configBundle corev1.Secret
+	if err := r.Get(ctx, types.NamespacedName{Namespace: quay.GetNamespace(), Name: quay.Spec.ConfigBundleSecret}, &configBundle); err != nil {
+		log.Error(err, "unable to retrieve referenced `configBundleSecret`", "configBundleSecret", quay.Spec.ConfigBundleSecret)
+
+		return ctrl.Result{}, err
+	}
+
+	log.Info("successfully retrieved referenced `configBundleSecret`", "configBundleSecret", configBundle.GetName())
```
Comment (josephschorr) [inline controllers/quayregistry_controller.go:65]: Might be worth outputting a hash or ID of the found secret to help coordinate changes

File: api/v1/groupversion_info.go (hunk 1 above)
Comment (alecmerdler) [reply api/v1/groupversion_info.go:29]: We can still change this, so let's discuss this in another pull request.

File: controllers/quayregistry_controller.go (hunk 2 above)
Comment (alecmerdler) [reply controllers/quayregistry_controller.go]: Good idea, I added the `resourceVersion` to the log output.

**Pull request merged**